addons/binding/org.openhab.binding.jeelink/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,224 @@
+			<channel id="applianceTime" typeId="appliance-time-channel" />
+			<channel id="sensorTime" typeId="sensor-time-channel" />
+			<channel id="resets" typeId="resets-channel" />
+		</channels>
+
+		<config-description>
+			<parameter name="sensorId" type="text" required="true">
+                <label>@text/parameter.sensorid.label</label>
+                <description>@text/parameter.sensorid.description</description>
+			</parameter>
+			<parameter name="updateInterval" type="integer" required="true" min="0" max="3600" unit="s" step="5">
+                <label>@text/parameter.updateinterval.label</label>
+                <description>@text/parameter.updateinterval.description</description>
+				<default>60</default>
+			</parameter>
+			<parameter name="sensorTimeout" type="integer" required="true" min="5" max="3600" unit="s" step="5">
+                <label>@text/parameter.sensortimeout.label</label>
+                <description>@text/parameter.sensortimeout.description</description>
+				<default>60</default>
+			</parameter>
+			<parameter name="bufferSize" type="integer" required="true" min="0" max="100">
+                <label>@text/parameter.buffersize.label</label>
+                <description>@text/parameter.buffersize.description</description>
+				<default>20</default>
+			</parameter>
+		</config-description>
+	</thing-type>
+
+	<!-- Current Watt Channel Type -->
+	<channel-type id="current-watt-channel">
+		<item-type>Number</item-type>
+		<label>@text/channel-type.current-watt-channel.label</label>
+		<description>@text/channel-type.current-watt-channel.description</description>
+		<state readOnly="true" />
+	</channel-type>
+
+	<!-- Max Watt Channel Type -->
+	<channel-type id="max-watt-channel">
+		<item-type>Number</item-type>
+		<label>@text/channel-type.max-watt-channel.label</label>
+		<description>@text/channel-type.max-watt-channel.description</description>
+		<state readOnly="true" />
+	</channel-type>
+
+	<!-- Total Consumption Channel Type -->
+	<channel-type id="consumption-total-channel">
+		<item-type>Number</item-type>
+		<label>@text/channel-type.consumption-total-channel.label</label>
+		<description>@text/channel-type.consumption-total-channel.description</description>
+		<state readOnly="true" />
+	</channel-type>
+
+	<!-- Appliance On Time Channel Type -->
+	<channel-type id="appliance-time-channel">
+		<item-type>Number</item-type>
+		<label>@text/channel-type.appliance-time-channel.label</label>
+		<description>@text/channel-type.appliance-time-channel.description</description>
+		<state readOnly="true" />
+	</channel-type>
+
+	<!-- Sensor On Time Channel Type -->
+	<channel-type id="sensor-time-channel">
+		<item-type>Number</item-type>
+		<label>@text/channel-type.sensor-time-channel.label</label>
+		<description>@text/channel-type.sensor-time-channel.description</description>
+		<state readOnly="true" />
+	</channel-type>
+
+	<!-- Resets Channel Type -->
+	<channel-type id="resets-channel">
```

Comment:
If you really think this is needed (what does it do?), have you at least considered to mark it "advanced=true"?